### PR TITLE
HLE: Move NS:S into APT and remove NS

### PR DESCRIPTION
**Pull request by jroweboy**
wwylele mentioned that these two services should be combined.

As a question: I'm not sure if making NS_S extend APT's module is really what i should be doing here. And also MaxAPTSessions seems weird to put there. please lemme know what you think.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/4472)

### CODE REVIEW

**Comment by wwylele** [comment]
Letting NS extending APT interface class is not the right way to go, as they doesn't share the same or similar service functions. You can make two base classes `APT::Module::APTInterface` and `APT::Module::NSInterface`, and let APT and NS extend them respectively. The old value for NS max sessions `2` should be kept unless proven wrong by hwtest/disassembly.

**Comment by jroweboy** [comment]
Updated. Is this more in line with what you were expecting?

**Comment by wwylele** [comment]
@jroweboy Yes, though you don't need a `BaseInterface` and do multi-inheritance. Just let `NSInterface` and `APTInterface` store the module pointer on their own. Also, Move `LaunchTitle` to its own `ns.cpp` file. This file can get really big in the future.

File: src/core/hle/service/apt/ns.h [hunk 1]
```diff
@@ -17,7 +17,4 @@ namespace Service::NS {
 /// Loads and launches the title identified by title_id in the specified media type.
 Kernel::SharedPtr<Kernel::Process> LaunchTitle(FS::MediaType media_type, u64 title_id);
 
-/// Registers all NS services with the specified service manager.
-void InstallInterfaces(Core::System& system);
-
-} // namespace Service::NS
+} // namespace Service::NS
```
Comment (wwylele) [inline src/core/hle/service/apt/ns.h]: New line at the end

File: src/core/hle/service/apt/ns_s.cpp [hunk 2]
```diff
@@ -2,11 +2,12 @@
...
 // Refer to the license.txt file included.
 
-#include "core/hle/service/ns/ns_s.h"
+#include "core/hle/service/apt/ns_s.h"
 
 namespace Service::NS {
 
-NS_S::NS_S() : ServiceFramework("ns:s", 2) {
+NS_S::NS_S(std::shared_ptr<Service::APT::Module> apt)
+    : Service::APT::Module::NSInterface(std::move(apt), "NS:S", Service::APT::MaxAPTSessions) {
```
Comment (Subv) [inline src/core/hle/service/apt/ns_s.cpp]: leave the 2 instead of MaxAPTSessions

**Comment by jroweboy** [comment]
All reviews addressed and also fixed a bug where I improperly capitalized NS:S when its supposed to be lower case ns:s

File: src/core/hle/service/apt/apt.h [hunk 3]
```diff
@@ -592,8 +602,8 @@ class Module final {
         void CheckNew3DS(Kernel::HLERequestContext& ctx);
 
     private:
-        std::shared_ptr<Module> apt;
         bool application_reset_prepared{};
+        std::shared_ptr<Module> apt;
```
Comment (zhaowenlan1779) [inline src/core/hle/service/apt/apt.h]: why is this moved?

File: src/core/hle/service/apt/apt.cpp [hunk 4]
```diff
@@ -12,12 +12,14 @@
 #include "core/hle/kernel/mutex.h"
 #include "core/hle/kernel/shared_memory.h"
 #include "core/hle/romfs.h"
+#include "core/hle/service/am/am.h"
```
Comment (zhaowenlan1779) [inline src/core/hle/service/apt/apt.cpp]: is this used?

File: src/core/hle/service/apt/apt.h [hunk 5]
```diff
@@ -9,6 +9,7 @@
 #include "common/common_types.h"
 #include "common/swap.h"
 #include "core/hle/kernel/kernel.h"
+#include "core/hle/service/fs/archive.h"
```
Comment (zhaowenlan1779) [inline src/core/hle/service/apt/apt.h]: ditto

**Comment by wwylele** [comment]
What's the status of this?

**Comment by B3n30** [comment]
@jroweboy Are you planning on continue with that PR and fix the conflicts?

**Comment by jroweboy** [comment]
Rebased and updated
